developer-workflow/sbom.rst

```diff
@@ -0,0 +1,109 @@
+  (`CPE <https://nvd.nist.gov/products/cpe>`_ and
+  `Package URLs <https://github.com/package-url/purl-spec/blob/master/PURL-SPECIFICATION.rst>`_
+  or "PURLs")
+* Path(s) to include and exclude in the CPython source tree corresponding to this dependency.
+
+After gathering this information:
+
+1. Add the information into a new entry in ``packages`` in the file
+   :cpy-file:`Misc/sbom.spdx.json`. Don't worry about formatting, the tool will
+   auto-format your manually written JSON. The fields to fill out include:
+
+   * ``name`` for the project name.
+   * ``SPDXID`` which will be ``"SPDXRef-PACKAGE-{name}"``
+   * ``licenseConcluded`` for the SPDX license identifier of the project license.
+   * ``versionInfo`` for the version of the project.
+   * ``downloadLocation`` should be an HTTPS URL for the project download as an archive.
+   * ``checksums[0].checksumValue`` and ``.algorithm`` will be the SHA-256
+     checksum of the downloaded archive.
+   * ``originator`` for the original author information, prefix with either an
+     ``Organization:`` or ``Person:`` depending on the author/maintenance situation.
+   * ``primaryPackagePurpose`` will likely be ``"SOURCE"``.
+   * ``externalReferences`` is a list of one or more project identifiers,
+     either CPE or Package URL. The value for ``referenceLocator`` must include
+     the value in ``versionInfo`` to ensure the identifier
+     corresponds to the correct release of the software. You can read more about
+     external references in the `SPDX SBOM specification`_.
+2. If a new license ID is to be used, add the license expression to
+   ``ALLOWED_LICENSE_EXPRESSIONS`` in the :cpy-file:`Tools/build/generate_sbom.py`.
+3. Add the paths to include and exclude into a ``PackageFiles`` instance
+   with a key corresponding to the SBOM ID for the package (``SPDXID`` without the
+   ``SPDXRef-PACKAGE-*`` prefix) in :cpy-file:`Tools/build/generate_sbom.py`.
+4. Run the tool with ``make regen-sbom`` or ``python Tools/build/generate_sbom.py``.
+   Ensure that the tool doesn't fail with any validation errors.
+5. Compare the changes to :cpy-file:`Misc/sbom.spdx.json` with ``$ git diff``, check
+   that all information appears correct.
+6. Commit the changes to :cpy-file:`Misc/sbom.spdx.json` and
+   :cpy-file:`Tools/build/generate_sbom.py`.
+
+.. _SPDX SBOM specification: https://spdx.github.io/spdx-spec/v2-draft/external-repository-identifiers/
+
+Removing a dependency
+---------------------
+
+When removing a dependency:
+
+1. Remove the entry from the :cpy-file:`Misc/sbom.spdx.json`
+   under the ``packages`` field.
+2. Remove the corresponding ``PackageFiles`` entry in :cpy-file:`Tools/build/generate_sbom.py`
+3. Run the tool with ``make regen-sbom`` or ``python Tools/build/generate_sbom.py``.
+   Ensure that the tool doesn't fail with any validation errors.
+4. Compare the changes to :cpy-file:`Misc/sbom.spdx.json` with ``$ git diff``, check
```

Comment:
```suggestion
4. Compare the changes to :cpy-file:`Misc/sbom.spdx.json` with ``git diff``, check
```

Reply:
I was just about to suggest the same, but you beat me to it :)